akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -522,14 +558,13 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
           behavior(setup, newState)
 
         case RemoveTick =>
-          // ok to update from several nodes but more efficient to try to do it from one node
-          if (isLeader) {
+          // ok to update from several nodes but more efficient to not do the same from all
+          if (ThreadLocalRandom.current().nextInt(state.registry.nodes.size) == 0) {
```

Comment:
Can't rely on leader here, because a completely new node may be the leader (lowest address) and then it doesn't know about `removedNodes`. Spreading the responsibility of the cleanup over all nodes.

Reply:
no, this isn't completely safe. It's possible that all of the remaining nodes that know about the removedNodes are also removed. That may result in entries that are not removed by this RemoveTick process.

I have to take this for another round.